python-heasoft-osa/Dockerfile

```
@@ -0,0 +1,59 @@
+# For finding latest versions of the base image see
+# https://github.com/SwissDataScienceCenter/renkulab-docker
+#ARG RENKU_BASE_IMAGE=renku/renkulab-py:3.7-0.7.3
+#FROM ${RENKU_BASE_IMAGE}
+FROM integralsw/osa-python:11.1-16-g88c002b7-20210507-170349-refcat-43.0-heasoft-6.29-python-3.8.2
+
+# Uncomment and adapt if code is to be included in the image
+# COPY src /code/src
+
+# Uncomment and adapt if your R or python packages require extra linux (ubuntu) software
+# e.g. the following installs apt-utils and vim; each pkg on its own line, all lines
+# except for the last end with backslash '\' to continue the RUN line
+#
+# USER root
+# RUN apt-get update && \
+#    apt-get install -y --no-install-recommends \
+#    apt-utils \
+#    vim
+# USER ${NB_USER}
+
+# install the python dependencies
+#COPY requirements.txt environment.yml /tmp/
+#RUN conda env update -q -f /tmp/environment.yml && \
+#    /opt/conda/bin/pip install -r /tmp/requirements.txt && \
+#    conda clean -y --all && \
+#    conda env export -n "root"
+
+# RENKU_VERSION determines the version of the renku CLI
+# that will be used in this image. To find the latest version,
+# visit https://pypi.org/project/renku/#history.
+ARG RENKU_VERSION=2.7.0
+
+########################################################
+# Do not edit this section and do not add anything below
+
+RUN export HOME_OVERRRIDE=/tmp && source /init.sh && \
```

why are you overriding the home directory? Also, the `init.sh` script is missing?

---

> why are you overriding the home directory? Also, the `init.sh` script is missing?

`init.sh` is in the [base image](https://github.com/ISDC-integral/osa-python-docker/blob/master/init.sh).
HEASOFT requires writable home to initialize, even during container build.

However, here, we could do just this:

```suggestion
RUN .  /etc/pyenvrc && \
```

---

ah right, I forgot you are not using a renku base image - have you tried that this works with renkulab?

---

It used to work. Now with the last change it does not, it seems. We'll fix.

---

couldn't you add the user above this line and just use their normal home directory?